pkg/controller/podgc/gc_controller.go

```diff
@@ -298,7 +306,23 @@ func (o byCreationTimestamp) Less(i, j int) bool {
 	return o[i].CreationTimestamp.Before(&o[j].CreationTimestamp)
 }
 
-func (gcc *PodGCController) deletePod(ctx context.Context, namespace, name string) error {
+func (gcc *PodGCController) deletePodWithCondition(ctx context.Context, namespace, name string, condition *v1.PodCondition) error {
```

Comment:
```suggestion
func (gcc *PodGCController) failAndDeletePodWithCondition(ctx context.Context, namespace, name string, condition *v1.PodCondition) error {
```

And probably beneficial to add a comment.

Reply:
I have applied the change for now, but I think it would be better to fix this issue (and set the phase=Failed) in a dedicated PR / KEP, so that we can also simplify the job_controller logic.